4 - AnimalShelter/Code/Animal.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+
+
+namespace AnimalShelter.Code
+{
+    public class Animal
+    {
+        public Guid UniqueAnimalId { get; private set; }
+        public AnimalType AnimalType { get; set; }
+        public Animal(AnimalType animalType)
```

Comment:
I really like the way you built the animal object. Clean, easy to read and create and is single responsibility.